src/components/FormComponents/Sykmeldingsperiode/PeriodePicker.tsx

```diff
@@ -1,97 +1,82 @@
+    const { field: fromField, fieldState: fromFieldState } = useController<SykmeldingFormValues, FormField>({
+        name: `${name}.fom`,
+        rules: { required: 'Du må fylle inn fra dato.' },
+    });
+    const { field: toField, fieldState: toFieldState } = useController<SykmeldingFormValues, FormField>({
+        name: `${name}.tom`,
+        rules: { required: 'Du må fylle inn til dato.' },
     });
 
-    useEffect(() => {
-        if (datepickerInputProps.value && datepickerInputProps.value.toString() !== field.value?.toString()) {
-            field.onChange(datepickerInputProps.value);
-        } else if (!datepickerInputProps.value && field.value) {
-            field.onChange(null);
-        }
-    }, [datepickerInputProps.value, field]);
-
-    return (
-        <div>
-            <UNSAFE_DatePicker.Input
-                id={field.name}
-                {...datepickerInputProps}
-                label={label}
-                placeholder="DD.MM.ÅÅÅÅ"
-                onFocus={(event) => {
-                    event.preventDefault();
-                }}
-            />
-            <FieldError error={fieldState.error} />
-        </div>
-    );
-}
-
-function PeriodePicker({ index }: PeriodePickerProps): JSX.Element {
-    const { datepickerProps, toInputProps, fromInputProps, reset } = UNSAFE_useRangeDatepicker({
+    const { datepickerProps, toInputProps, fromInputProps, selectedRange, setSelected } = UNSAFE_useRangeDatepicker({
         today: new Date(),
         defaultSelected: {
-            from: undefined,
-            to: undefined,
+            from: fromField.value,
+            to: toField.value,
         },
     });
 
-    const fomName: FomName = `periode.${index}.fom`;
-    const tomName: TomName = `periode.${index}.tom`;
+    const onChangeFrom = fromField.onChange;
+    const onChangeTo = toField.onChange;
+    const hasMounted = useRef(false);
+    useEffect(() => {
+        if (!hasMounted.current) {
+            hasMounted.current = true;
+            return;
+        }
+
+        onChangeFrom(selectedRange?.from);
+        onChangeTo(selectedRange?.to);
+    }, [selectedRange, onChangeFrom, onChangeTo]);
 
     return (
         <>
             <div className={styles.periodePicker}>
                 <UNSAFE_DatePicker {...datepickerProps}>
-                    <DatepickerInput
-                        name={fomName}
+                    <UNSAFE_DatePicker.Input
+                        ref={fromField.ref}
+                        id={fromField.name}
+                        {...fromInputProps}
                         label="Fra"
-                        datepickerInputProps={fromInputProps}
-                        error="Du må fylle inn fra dato."
+                        placeholder="DD.MM.ÅÅÅÅ"
+                        onFocus={(event) => {
+                            event.preventDefault();
+                        }}
+                        error={fromFieldState.error?.message}
```

Comment:
Kanskje vi skal bytte ut `FieldError` med `error` prop for dei andre DS komponentane også etterkvart?

Reply:
ah okey, vi ser då.